core-bundle/src/Controller/BackendPreviewSwitchController.php

```diff
@@ -155,13 +161,19 @@ private function authenticatePreview(Request $request): void
         $showUnpublished = 'hide' !== $request->request->get('unpublished');
 
         if (null !== $frontendUsername) {
-            $this->previewAuthenticator->authenticateFrontendUser($frontendUsername, $showUnpublished);
+            if (!$this->previewAuthenticator->authenticateFrontendUser($frontendUsername, $showUnpublished)) {
+                $message = $this->translator->trans('ERR.previewSwitchInvalidUsername', [$frontendUsername], 'contao_default');
+
+                return new Response($message, Response::HTTP_BAD_REQUEST);
```

Comment:
This is wrong. Bad request is to be used if a request was syntactically incorrect and not understood by the server, which is not the case here. If a user is attempting to authenticate, but provides invalid credentials, the response should have a status of 401.

Reply:
I don't think that would not be correct. This already is an authenticated request and thus 401 would indicate that you are not authenticated (which you are).

Reply:
But the request is already authenticated and 401 would indicate to you that you are not authenticated - but you are authenticated. 

Also we need to distinguish between not authenticated and invalid username for the JavaScript to work. Currently we only rely on status codes.

Reply:
As discussed in the Contao call, we want to use status 422 here.

Reply:
Implemented in 77a7ad6fa